--- a/lib/native/base/base.cc
+++ b/lib/native/base/base.cc
@@ -2951,7 +2951,7 @@
 static StatResult StatAt(int fd, bool fd_is_directory, const char *filename, unsigned int flags, FileInfo *out_info)
 {
-#if defined(__linux__)
+#if defined(__linux__) && !defined(__ANDROID__)
     const char *pathname = filename;
     int stat_flags = (flags & (int)StatFlag::FollowSymlink) ? 0 : AT_SYMLINK_NOFOLLOW;
